nrf52840-hal/src/ieee802154.rs

```diff
@@ -243,6 +259,50 @@ impl<'c> Radio<'c> {
             .write(|w| w.txpower().variant(power._into()));
     }
 
+    /// Sample the received signal power (i.e. the presence of possibly interfering signals)
+    /// within the bandwidth of the currently used channel for sample_cycles iterations.
+    /// The result can be used to find a suitable ED threshold for Energy Detection-based CCA mechanisms.
+    ///
+    /// Returns the *maximum* measurement recorded during sampling as reported by the hardware (not in dBm!)
```

Comment:
I would swap these last two sentences so that it is clearer that 1 iteration = RSSI sample; RSSI sampling takes 128 us; the RSSI value is averaged over the measurement time; but then *N* iterations of the RSSI sample are performed; and the *maximum* RSSI value among those N iterations is returned.